File: backend/dotnet/Models/Notification.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace dotnet.Models
{
    [Table("notifications")]
    public class Notification
    {
        [Key]
        [Required]
        [Column("notification_id")]
        public string Id { get; set; } = string.Empty;

        [Required]
        [Column("type")]
        public string Type { get; set; } = string.Empty;

        [Required]
        [Column("title")]
        public string Title { get; set; } = string.Empty;

        [Required]
        [Column("content")]
        public string Content { get; set; } = string.Empty;

        [Column("read")]
        [DefaultValue(false)]
        public bool Read { get; set; } = false;

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("link")]
        public string? Link { get; set; }

        // Navigation property
        public ICollection<UserNotification> UserNotifications { get; set; } = new List<UserNotification>();
    }
}

File: backend/golang/internal/model/notification.go
package model

import "time"

type Notification struct {
	ID        string    `json:"id" gorm:"primaryKey"`
	Type      string    `json:"type" gorm:"not null"`
	Title     string    `json:"title" gorm:"not null"`
	Content   string    `json:"content"`
	Read      bool      `json:"read" gorm:"default:false"`
	Link      string    `json:"link"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`

	Users []UserNotification `gorm:"foreignKey:NotificationID" json:"userNotifications"`
}
